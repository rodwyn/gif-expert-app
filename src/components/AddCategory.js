import React, { useState } from 'react'

export const AddCategory = () => {
  const [inputValue, setInputValue] = useState('');
  const handleInputChange = event => {
    setInputValue(event.target.value);
  };
  const handleSubmit = event => { event.preventDefault(); };
  return (
    <form onSubmit={handleSubmit}>
      <input type="text" value={inputValue} onChange={handleInputChange} />
    </ form>
  )
}
