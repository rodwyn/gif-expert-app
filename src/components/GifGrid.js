import React, {useState, useEffect} from 'react'
import { GifGridItem } from './GifGridItem'

export const GifGrid = ({category}) => {
  const [images, setImages] = useState([]);

  useEffect(()=>{
    getGifs();
  }, []);

  const getGifs = async() => {
    const url = 'https://api.giphy.com/v1/gifs/search?q=gorillaz&limit=10&api_key=swbFspRTjwsaFOAtITNuUWGQ0AQGOedH';
    const response = await fetch(url);
    const {data} = await response.json();
    const gifs = data.map( img => {
      return {
        id: img.id,
        title: img.title,
        url: img.images?.downsized_medium.url
      }
    });

    console.log(gifs);
    setImages(gifs);
  };

  return (
    <>
      <h3>{category}</h3>
      <div className="card-grid">  
        {
          images.map(
            img => 
              <GifGridItem key={img.id} {...img} />
          )
        }
      </div>
    </>
  )
}
