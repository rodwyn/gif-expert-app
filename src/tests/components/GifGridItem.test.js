import React from 'react';
import {shallow} from 'enzyme';
import {GifGridItem} from '../../components/GifGridItem';

describe('Testing <GifGridItem> component.', ()=> {
  const title = 'my title';
  const url = 'https://rodwyn.com/icon.jpg';

  test('Should render the component correctly.', ()=>{
    const wrapper = shallow(<GifGridItem title={title} url={url} />);
    expect(wrapper).toMatchSnapshot();
  });
});