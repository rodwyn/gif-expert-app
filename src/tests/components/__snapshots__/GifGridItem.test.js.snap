// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <GifGridItem> component. Should render the component correctly. 1`] = `
<div
  className="card animate__animated animate__bounceIn"
>
  <img
    alt="my title"
    src="https://rodwyn.com/icon.jpg"
  />
  <p>
    my title
  </p>
</div>
`;
